storm-core/src/jvm/org/apache/storm/metrics2/StormMetricRegistry.java

```diff
@@ -0,0 +1,156 @@
+                StormMetricRegistry.gauge(0.0, name + "-sojourn-time-ms", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0L, name + "-overflow", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0.0F, name + "-percent-full", topologyId, componentId, port)
+        );
+    }
+
+    public static Meter meter(String name, WorkerTopologyContext context, String componentId, String executorId, String streamId){
+        String metricName = metricName(name, context.getStormId(), componentId, streamId,executorId, context.getThisWorkerPort());
+        return REGISTRY.meter(metricName);
+    }
+
+    public static Counter counter(String name, WorkerTopologyContext context, String componentId, String executorId, String streamId){
+        String metricName = metricName(name, context.getStormId(), componentId, streamId,executorId, context.getThisWorkerPort());
+        return REGISTRY.counter(metricName);
+    }
+
+    public static void start(Map<String, Object> stormConfig, DaemonType type){
+        String localHost = "localhost";
+        try {
+            hostName = dotToUnderScore(Utils.localHostname());
+        } catch (UnknownHostException e) {
+             LOG.warn("Unable to determine hostname while starting the metrics system. Hostname will be reported" +
+                     " as 'localhost'.");
+        }
+
+        LOG.info("Starting metrics reporters...");
+        List<Map<String, Object>> reporterList = (List<Map<String, Object>>)stormConfig.get(Config.STORM_METRICS_REPORTERS);
+        if(reporterList != null && reporterList.size() > 0) {
+            for (Map<String, Object> reporterConfig : reporterList) {
+                // only start those requested
+                List<String> daemons = (List<String>) reporterConfig.get("daemons");
+                for (String daemon : daemons) {
+                    if (DaemonType.valueOf(daemon.toUpperCase()) == type) {
+                        startReporter(stormConfig, reporterConfig);
+                    }
+                }
+            }
+        }
+    }
+
+    public static MetricRegistry registry(){
+        return REGISTRY;
+    }
+
+    private static void startReporter(Map<String, Object> stormConfig, Map<String, Object> reporterConfig){
+        String clazz = (String)reporterConfig.get("class");
+        StormReporter reporter = null;
+        LOG.info("Attempting to instantiate reporter class: {}", clazz);
+        try{
+            reporter = (StormReporter)Metrics2Utils.instantiate(clazz);
+        } catch(Exception e){
+            LOG.warn("Unable to instantiate metrics reporter class: {}. Will skip this reporter.", clazz, e);
+        }
+        if(reporter != null){
+            reporter.prepare(REGISTRY, stormConfig, reporterConfig);
+            reporter.start();
+            REPORTERS.add(reporter);
+        }
+
+    }
+
+
+    public static void stop(){
```

Comment:
Why storm and not close ala AutoCloseable? is it because codahale uses stop?

Reply:
Yes.